src/effects/builtin/noisereduction/NoiseReductionSlider.qml

```diff
@@ -0,0 +1,45 @@
+import QtQuick
+import QtQuick.Layouts
+import Muse.UiComponents
+import Audacity.Effects
+
+Row {
+    id: root
+
+    property alias value: slider.value
+    property bool isInt: false
+    property alias from: slider.from
+    property alias to: slider.to
+    property alias measureUnitsSymbol: textControl.measureUnitsSymbol
+
+    QtObject {
+        id: prv
+        property int decimals: root.isInt ? 0 : 2
+    }
+
+    spacing: 16
+
+    StyledSlider {
+        id: slider
+        width: (root.width - root.spacing) * 0.6
+        anchors.verticalCenter: root.verticalCenter
+        onMoved: {
+            slider.value = prv.decimals === 0 ? Math.round(slider.value) : parseFloat(slider.value.toFixed(prv.decimals));
```

Comment:
This is a bit problematic. If you assign the value to a property, you break the property binding. 
After this, the binding model -> slider.value is destroyed, and when the model emits propertyChanged() signal, the value will not be updated.